ddtrace/debugging/_exception/auto_instrument.py

```diff
@@ -72,31 +69,41 @@ def on_span_finish(self, span):
             # No error to capture
             return
 
-        _type, _exc, _tb = sys.exc_info()
+        _, _, _tb = sys.exc_info()
         if _tb is None or _tb.tb_frame is None:
             # If we don't have a traceback there isn't much we can do
             return
 
-        seq_nr = count()
+        seq = count(1)
         exc_id = uuid.uuid4()
 
         # DEV: We go from the handler up to the root exception
         while _tb:
             frame = _tb.tb_frame
-            n = next(seq_nr)
+            code = frame.f_code
+            seq_nr = next(seq)
 
             snapshot = SpanExceptionSnapshot(
                 probe=SpanExceptionProbe.build(exc_id, frame),
                 frame=frame,
                 thread=current_thread(),
                 trace_context=span,
                 exc_id=exc_id,
-                seq_nr=n,
+                seq_nr=seq_nr,
             )
 
+            # Add correlation tags on the span
+            span.set_tag_str("_dd.debug.error.%d.snapshot.id" % seq_nr, snapshot.uuid)
+            span.set_tag_str("_dd.debug.error.%d.function" % seq_nr, code.co_name)
+            span.set_tag_str("_dd.debug.error.%d.file" % seq_nr, code.co_filename)
+            span.set_tag_str("_dd.debug.error.%d.line" % seq_nr, str(frame.f_lineno))
+
             # Capture
             snapshot.line()
 
             self.collector.push(snapshot)
 
             _tb = _tb.tb_next
+
+        span.set_tag_str("error.debug.info-captured", "true")
```

Comment:
I've updated the RFC recently to make this `debug-info-captured`, since the other changes have meant that there is no longer any reason to add the extra nesting level of "error.debug.". Does that make sense to you?

Reply:
👍  Yep. I'll update this.